src/main/java/fr/moribus/imageonmap/PluginConfiguration.java

```diff
@@ -21,17 +21,101 @@
 import fr.zcraft.zlib.components.configuration.Configuration;
 import fr.zcraft.zlib.components.configuration.ConfigurationItem;
 
+import java.io.*;
 import java.util.Locale;
 
 import static fr.zcraft.zlib.components.configuration.ConfigurationItem.item;
 
 
 public final class PluginConfiguration extends Configuration
```

Comment:
The whole changes on this file should be rollbacked.

You shouldn't use a line-per-line read counting lines to parse a configuration file. It's pretty ugly and would break at the first blank line added by a user.

Plus, zLib's [configuration manager](https://github.com/zDevelopers/zLib/wiki/Configuration) is way more powerful.

Reply:
I at some point seem to have changed it, and this is what I have now:

```public static void initialize() throws IOException {
		File f = new File(ImageOnMap.getPlugin().getDataFolder(), "config.yml");
		ArrayList<String> sa = readFileToString(f);
		for(int i = 0; i < sa.size(); i++) {
			String s = sa.get(i);
			if(!s.startsWith("#")){
				if(s.contains("lang")) {
					s = parseYAML("lang");
					if(s.trim() == "en_US") LANG = Locale.ENGLISH;
					if(s.trim() == "fr_FR") LANG = Locale.FRENCH;
					else LANG = Locale.ENGLISH;
				}

				if(s.contains("collect-data")) {
					s=parseYAML(s);

					if(s.trim() == "true") COLLECT_DATA = true;
					else COLLECT_DATA = false;
				}
				if(s.contains("map-global-limit")) {
					s = parseYAML(s);
					MAP_GLOBAL_LIMIT = Integer.parseInt(s);
				}
				if(s.contains("map-player-limit")) {
					s = parseYAML(s);
					MAP_PLAYER_LIMIT = Integer.parseInt(s);
				}
				if(s.contains("limit-map-size-x")) {
					s = parseYAML(s);
					LIMIT_SIZE_X= Integer.parseInt(s);
				}
				if(s.contains("limit-map-size-y")) {
					s = parseYAML(s);
					LIMIT_SIZE_Y = Integer.parseInt(s);
				}

				if(s.contains("save-full-image")) {
					s = parseYAML(s);
					if(s.trim().contains("true")) SAVE_FULL_IMAGE = true;
					else SAVE_FULL_IMAGE = false;
				}
			}
		}
	}```

Reply:
Is that better? it should also be more pretty :p

Reply:
We wrote a whole fully tested component to handle configuration in zLib and it works well, there is no need to reinvent the wheel, with a less-optimal option. Our way to load the configuration works, just use it! You only need an environment with Maven, for the dependencies to be loaded correctly, but the plugin won't compile without Maven anyway.

Reply:
I did a bit of testing, and ImageOnMap does not read the configuration. For instance, collect-data always returns true, no matter what.

Reply:
I believe that the PluginConfiguration reads from the DEFAULT config.yml file instead of the config.yml in the plugins folder. I set up a makeshift debug command that tells me values, and this is what I get.

For this config.yml:
```###  ImageOnMap configuration file


# Plugin language. Empty: system language.
# Available: en_US (default, fallback) and fr_FR.
lang:


# Allows collection of anonymous statistics on plugin environment and usage
# The statistics are publicly visible here: http://mcstats.org/plugin/ImageOnMap
collect-data:false


# Images rendered on maps consume Minecraft maps ID, and there are only 32 767 of them.
# You can limit the maximum number of maps a player, or the whole server, can use with ImageOnMap.
# 0 means unlimited.
map-global-limit: 0
map-player-limit: 0


#Maximum size in pixels for an image to be. 0 is unlimited.
limit-map-size-x: 256
limit-map-size-y: 512


#Should the full image be saved when a map is rendered?
save-full-image: true```
I get
![screenshot_1](https://user-images.githubusercontent.com/12013191/37752158-dbd29434-2d6c-11e8-855b-307193f02b26.png)

which matches default config.yml
```###  ImageOnMap configuration file


# Plugin language. Empty: system language.
# Available: en_US (default, fallback) and fr_FR.
lang:


# Allows collection of anonymous statistics on plugin environment and usage
# The statistics are publicly visible here: http://mcstats.org/plugin/ImageOnMap
collect-data: true 
NOTE: collect-data was false in the default-config when I tested 

# Images rendered on maps consume Minecraft maps ID, and there are only 32 767 of them.
# You can limit the maximum number of maps a player, or the whole server, can use with ImageOnMap.
# 0 means unlimited.
map-global-limit: 0
map-player-limit: 0


#Maximum size in pixels for an image to be. 0 is unlimited.
limit-map-size-x: 0
limit-map-size-y: 0


#Should the full image be saved when a map is rendered?
save-full-image: false```